app/models/spree/product_decorator.rb

```diff
@@ -1,4 +1,9 @@
 Spree::Product.class_eval do
+
+  def variants_in_stock
```

Comment:
I still don't think this is needed, same amount of typing `variants.in_stock` vs `variants_in_stock`, it also adds extra level to stack trace.